mlir/lib/IR/AffineExpr.cpp

```diff
@@ -117,6 +117,122 @@ bool AffineExpr::isSymbolicOrConstant() const {
   llvm_unreachable("Unknown AffineExpr");
 }
 
+bool AffineExpr::isSymbolic() const {
+  switch (getKind()) {
+  case AffineExprKind::SymbolId:
+    return true;
```

Comment:
You could just use a simple `if` to check and return true, and false for the rest.